cumulus/primitives/parachain-inherent/src/lib.rs

```diff
@@ -115,6 +117,9 @@ pub struct ParachainInherentData {
 	/// This information is used to ensure that a parachain node builds blocks
 	/// at a specified offset from the chain tip rather than directly at the tip.
 	pub relay_parent_descendants: Vec<RelayHeader>,
+	/// Contains the collator peer ID, which is later sent by the parachain to the
+	/// relay chain via a UMP signal to promote the reputation of the given peer ID.
+	pub collator_peer_id: Option<ApprovedPeerId>,
```

Comment:
@sw10pa just a dq, I am not sure about now, when we add this field here, doesn't it change encoding or with `None` no? I am thinking about when the runtime with this change is enacted, what about older validators/collators without this change, do they still work? Will the `ParachainInherentData` with `collator_peer_id: None` produce the same data as `ParachainInherentData` without `collator_peer_id` field? Will both validators/collators produce the same state_root or whatever? Other words, is this change backwards-compatible? Will the older validators/collators still work?